FIT-10

dxhoan420 opened the pull request:
Классы User и Role сделал, как и таблицы для них. Интересует мнение по реализации equals. Необходимо обсудить версионность файлов для changesets!
Предлагаю настройки Spring тоже перенести в yml файл.

Commit:
FIT-10 added users, roles, users_roles tables and entities. New config for Spring in application.yml

ws-eliseev commented on src/main/java/ws/eliseev/fitness/model/Role.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.Objects;
+import java.util.Set;
+
+@Entity @Table(name = "roles")
```
каждая аннотация с новой строчки

dxhoan420 replied:
OK

ws-eliseev commented on src/main/java/ws/eliseev/fitness/model/Role.java:
```diff
@@ -0,0 +1,39 @@
+public class Role {
+    @Id
+    // Need? @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column (name = "id")
```
имена колонок должны быть в верхнем регистре

dxhoan420 replied:
А на что это влияет?

ws-eliseev commented on src/main/java/ws/eliseev/fitness/model/User.java:
```diff
@@ -1,4 +1,57 @@
+    @JoinTable(name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    private Set<Role> roleSet;
```
в ManyToMany нужно объявить главную сущность

qrillich replied:
MappedBy

Commit:
FIT-10 added mappedBy attribute for Role, fixed columns names, remade annotation according to the CodeStyle

Commit:
FIT-10 Добавлено: создание последовательности в LB для user id pk; пол в сущности пользователя; аннотации для equals.
Переименованы названия таблиц, полей и файлов в соответсвии с кодстайлом.
Удалён лишний файл changelog, а его содержимое перемещено в оставшийся

Commit:
FIT-10 Инициализировал сеты ролей и пользователей хешсетом. Убрал equals и hashcode
